### PR TITLE
Tabula forms

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue

fixes {https://github.com/HausDAO/daohaus-monorepo/issues/747}

## Changes
- Create Publisher Form
- Create Publisher TX
- Create Article Form
- Create Article TX
- Field that fetches a pubId from the tabula subgraph and puts it in the form data
- TX builder arg type that pins JSON data to IPFS and returns a hash

## Next steps

See why Tabula doesn't render MD data 
Try to get Tabula frontend working for demo at MCON

## Packages Added

NA

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/core-app/src/legos/tx.ts [hunk 1]
```diff
@@ -270,3 +271,67 @@ export const TX: Record<string, TXLego> = {
...
+      {
+        contract: CONTRACT.POSTER,
+        method: 'post',
+        args: [
+          {
+            type: 'JSONDetails',
+            jsonSchema: {
+              action: { type: 'static', value: TABULA_TAGS.ARTICLE_ACTION },
+              publicationId: '.formValues.pubId',
+              article: { type: 'ipfsPinata', content: '.formValues.article' },
```
Comment (skuhlmann) [inline apps/core-app/src/legos/tx.ts:328]: looks to me like this needs to just be:

```
article: 'hash'
```
https://github.com/gnosis/tabula#create-article

**Review by skuhlmann — APPROVED** [review]
Approved with updated json schema in the article tx

File: apps/core-app/src/components/FakeMarkdown.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      }
+    }
+  }
+`;
+
+const handleFetchPubId = async (
+  safeId: string,
+  chainId: ValidNetwork,
+  setValue: UseFormSetValue<FieldValues>,
+
```
Comment (0xRowdy) [inline apps/core-app/src/components/FakeMarkdown.tsx:48]:
```suggestion
```

File: apps/core-app/src/components/FakeMarkdown.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    if (!shouldUpdate) return;
+    if (pubId) {
+      setValue(PUB_ID, pubId);
+      setFetchStatus(FetchStatus.Success);
+      return;
+    }
+    setFetchStatus(FetchStatus.NoPubId);
+    throw new Error(`No PubId found for safe address: ${safeId}`);
+  } catch (e) {
+    console.error(e);
```
Comment (0xRowdy) [inline apps/core-app/src/components/FakeMarkdown.tsx:73]:
```suggestion
```

File: libs/common-utilities/src/utils/ipfs.ts [hunk 4]
```diff
@@ -1,25 +1,38 @@
...
       },
-      body: formData,
+      body: jsonString,
     });
-    return response.json();
+    const data = await response.json();
+    if (typeof data?.IpfsHash === 'string') {
+      return data as JSONResponse;
+    } else {
+      console.log('response', response);
```
Comment (0xRowdy) [inline libs/common-utilities/src/utils/ipfs.ts:31]:
```suggestion
```

File: libs/common-utilities/src/utils/ipfs.ts [hunk 5]
```diff
@@ -1,25 +1,38 @@
...
+      throw new Error(`IPFS Pin failed.`);
+    }
   } catch (err) {
     console.error(err);
```
Comment (0xRowdy) [inline libs/common-utilities/src/utils/ipfs.ts:35]:
```suggestion
```

File: libs/tx-builder-feature/src/utils/ipfs.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  const res = await pinataPostJSON({
+    creds: {
+      pinata_api_key,
+      pinata_api_secret,
+    },
+    jsonString: JSON.stringify({ data: processedContent }),
+  });
+
+  const IPFSpin = res.IpfsHash;
+  console.log('***IPFSpin***', IPFSpin);
```
Comment (0xRowdy) [inline libs/tx-builder-feature/src/utils/ipfs.ts:49]:
```suggestion
```

File: apps/core-app/src/components/FakeMarkdown.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+  ReactSetter,
+  ValidNetwork,
+} from '@daohaus/common-utilities';
+
+import { gql, request } from 'graphql-request';
+import { useFormBuilder } from '@daohaus/haus-form-builder';
+import { FieldValues, UseFormSetValue } from 'react-hook-form';
+
+const PUB_ID = 'pubId';
+enum FetchStatus {
```
Comment (0xRowdy) [inline apps/core-app/src/components/FakeMarkdown.tsx:24]: Use a State Machine?
I'm not 100% clear on the logical flow from status to status and a state machine might not work here.

**Review by 0xRowdy — APPROVED** [review]
Looks like we should add to eslint ignore.
libs/dao-data/src/subgraph/schema-lens.generated.ts

**Pull request merged**